go/libraries/doltcore/env/environment.go

```diff
@@ -1251,17 +1254,75 @@ func (dEnv *DoltEnv) LockFile() string {
 	return f
 }
 
-// IsLocked returns true if this database's lockfile exists
+// IsLocked returns true if this database's lockfile exists and the pid contained in lockfile is alive.
 func (dEnv *DoltEnv) IsLocked() bool {
 	if dEnv.IgnoreLockFile {
 		return false
 	}
 
 	ok, _ := dEnv.FS.Exists(dEnv.LockFile())
-	return ok
+	if !ok {
+		return false
+	}
+
+	lockFilePid, err := getProcessFromLockFile(dEnv.FS, dEnv.LockFile())
+	if err != nil { // if there's any error assume that env is locked since the file exists
+		return true
+	}
+
+	// Check whether the pid that spawned the lock file is still running. Ignore it if not.
+	// For unix systems FindProcess will always return a nil err and a process object which is why we must use the checkPid
+	// function. For Windows systems FindProcess will correctly determine if a process exists or not.
+	p, err := os.FindProcess(lockFilePid)
+	if runtime.GOOS == "windows" {
```

Comment:
Instead of this, you should have different versions of this function in different files that only run on one particular platform

call it `processExists`

You might also check out if there's a lightweight library that does this

Reply:
I used this library https://github.com/mitchellh/go-ps. Gotta be good since it was made by the Hashicopr founder. Also tested in on windows